plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/AssignAnalysis.java

```diff
@@ -418,6 +418,15 @@ private TokensList getNonFunctionDefCompletionsFromAssign(ICodeCompletionASTMana
                 int unpackPos = -1;
                 boolean unpackBackwards = false;
                 if (assignDefinition != null) {
+                    // If we have a type, activation token has already been defined as type
+                    // So let's check first if we can find completions using the type. If we can not find, we just ignore it and continue normally
+                    if (assignDefinition.nodeType != null) {
```

Comment:
This doesn't seem like it's unpacking the type from a list when needed (asked for tests for this use-case).